components/concept/engine/src/main/java/mozilla/components/concept/engine/Engine.kt

```diff
@@ -96,6 +97,11 @@ interface Engine : WebExtensionRuntime, DataCleanable {
      */
     fun createSessionState(json: JSONObject): EngineSessionState
 
+    /**
+     * Creates a new [EngineSessionState] instances from the serialized JSON representation.
```

Comment:
nit: This needs an update: a new instance, but also needs to capture the new param.